packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2497,7 +2498,10 @@ export class SnapController extends BaseController<
       );
     }
 
-    if (permissionName === SnapEndowments.Rpc) {
+    if (
+      permissionName === SnapEndowments.Rpc ||
```

Comment:
@Mrtenz This might break if MetaMask calls it? How do I specify that I allow MM to call my snap as an origin?

Reply:
Good point. Should we allow MetaMask to bypass the check? 🤔

Reply:
Potentially? We could do it just for the keyring export so far, it is expected that MM can call that. I'm not sure what the expectation is for `onRpcRequest`, the expectation isn't really that it would be called by MM

Reply:
I changed it to always allow MetaMask for now. The extension can just look up an allowed origin regardless, and pretend to be that origin. 😅